zirgen/circuit/predicates/predicates.cpp

```diff
@@ -206,4 +224,39 @@ UnionClaim unionFunc(Assumption left, Assumption right) {
   return claim;
 }
 
+ReceiptClaim ReceiptClaim::fromRv32imV2(llvm::ArrayRef<Val>& stream) {
+  DigestVal input = readSha(stream);
+  Val isTerminate = readVal(stream);
+  DigestVal output = readSha(stream);
+  /*Val rng =*/readExtVal(stream);
+  /*Val shutdownCycle =*/readVal(stream);
+  DigestVal stateIn = readSha(stream);
+  DigestVal stateOut = readSha(stream);
```

Comment:
Is `readSha` correct considering that the memory image root is hashed with Poseidon2?

Reply:
@jbruestle thoughts? It's weird, but since the encoding of a sha digest and a poseidon2 digest are compatible, it doesn't seem to matter what the `DigestKind` gets set to. The issue, as I understand it, is that the recursion circuit's `IntoDigestOp` only supports sha-256. And since we're treating the poseidon2 digest as an opaque entity, this is safe to 'recast' it like this.

Reply:
Correct.  This isn't actually a SHA-256, but it is in fact encode identically, and should in fact be fine.